test/simulator/CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+  COMMAND ar q ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a $<TARGET_OBJECTS:bitbox_objects-simulator>
+  VERBATIM
+  COMMAND_EXPAND_LISTS
+)
+
+add_custom_target(bitbox_merged-simulator DEPENDS ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a)
+
+target_include_directories(
+  bitbox_objects-simulator
+  SYSTEM PUBLIC
+  ${SYSTEMINCLUDES}
+  ${CMAKE_SOURCE_DIR}/external
+  ${CMAKE_SOURCE_DIR}/external/ctaes
+  ${CMAKE_SOURCE_DIR}/external/fatfs/source
+  $<TARGET_PROPERTY:wallycore,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:secp256k1,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:fatfs,INTERFACE_INCLUDE_DIRECTORIES>
+)
+target_include_directories(
+  bitbox_objects-simulator
+  PUBLIC
+  ${INCLUDES}
+  ${CMAKE_CURRENT_SOURCE_DIR}/framework/includes
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_BINARY_DIR}/src
+)
+
+target_include_directories(
+  bitbox-simulator
+  SYSTEM PUBLIC
+  ${SYSTEMINCLUDES}
+  ${CMAKE_SOURCE_DIR}/external
+  ${CMAKE_SOURCE_DIR}/external/ctaes
+  ${CMAKE_SOURCE_DIR}/external/fatfs/source
+  $<TARGET_PROPERTY:wallycore,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:secp256k1,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:fatfs,INTERFACE_INCLUDE_DIRECTORIES>
+)
+
+target_include_directories(
+  bitbox-simulator
+  PUBLIC
+  ${INCLUDES}
+  ${CMAKE_CURRENT_SOURCE_DIR}/framework/includes
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_BINARY_DIR}/src
+)
+
+add_dependencies(bitbox_objects-simulator
+  rust-cbindgen
+  wallycore
+  secp256k1
+  ctaes
+  fatfs
+  sd-mock-simulator
+)
+target_link_libraries(bitbox-simulator PRIVATE ${LIBBITBOX02_RUST} "-lm")
+
+target_compile_definitions(bitbox_objects-simulator PUBLIC "PRODUCT_BITBOX_MULTI=1" "APP_BTC=1" "APP_LTC=1" "APP_U2F=1" "APP_ETH=1")
+target_compile_definitions(bitbox_objects-simulator PUBLIC TESTING)
+
+# Since wallycore is an external projects we need to specify the dependency
+add_dependencies(bitbox_objects-simulator libwally-core)
+
+target_compile_definitions(bitbox-simulator PUBLIC "PRODUCT_BITBOX_MULTI=1" "APP_BTC=1" "APP_LTC=1" "APP_U2F=1" "APP_ETH=1")
+target_compile_definitions(bitbox-simulator PUBLIC TESTING)
+
+target_link_libraries(bitbox-simulator
+  PUBLIC
+    secp256k1
+  PRIVATE
+    wallycore
+    ctaes
+    fatfs
+    sd-mock-simulator
+)
+
+if(SANITIZE_ADDRESS)
+  target_compile_options(bitbox_objects-simulator PUBLIC "-fsanitize=address")
+  target_compile_options(bitbox-simulator PUBLIC "-fsanitize=address")
+endif()
+if(SANTIZE_UNDEFINED)
+  target_compile_options(bitbox_objects-simulator PUBLIC "-fsanitize=undefined")
+  target_compile_options(bitbox-simulator PUBLIC "-fsanitize=undefined")
+endif()
+if(COVERAGE)
+  target_link_libraries(bitbox-simulator PUBLIC "--coverage")
+  target_compile_options(bitbox_objects-simulator PUBLIC "--coverage")
+  target_compile_options(bitbox-simulator PUBLIC "--coverage")
+endif()
```

Comment:
don't need coverage stuff here i think, as the simulator is not unit testing and not about seeing which lines are covered during execution